src/core/net/ip6_address.cpp

```diff
@@ -114,7 +114,7 @@ bool Address::IsRealmLocalAllMplForwarders(void) const
 bool Address::IsRoutingLocator(void) const
 {
     return (mFields.m16[4] == HostSwap16(0x0000) && mFields.m16[5] == HostSwap16(0x00ff) &&
-            mFields.m16[6] == HostSwap16(0xfe00));
+            mFields.m16[6] == HostSwap16(0xfe00) && (mFields.m8[14] & 0x02) == 0 && mFields.m8[14] < 0xfc);
```

Comment:
I found other functions in this file uses the same magic numbers. Shall I change them all?